build/builder.go

```diff
@@ -363,18 +363,33 @@ func (o *Options) IndexState() (IndexState, string) {
 		return IndexStateOption, fn
 	}
 
-	sortBranches(o.RepositoryDescription.Branches)
-	sortBranches(repo.Branches)
-
 	if o.IsDelta {
 		// TODO: Get rid of this guard once the delta shard behavior is the default
-		ignoreVersionOption := cmpopts.IgnoreFields(zoekt.RepositoryBranch{}, "Version")
+		existingBranchNames := make(map[string]struct{}, len(repo.Branches))
```

Comment:
`cmpopts.IgnoreFields(zoekt.RepositoryBranch{}, "Version")` doesn't seem to work with maps, so I just decided to allocate two separate ones.

Reply:
I pushed a commit that avoids using `cmp` which is a package made for testing. I don't think we need to incur the overhead.